src/Abp.ZeroCore.EntityFrameworkCore/EntityHistory/EntitySnapshotManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Domain.Entities;
+using Abp.Domain.Repositories;
+using Abp.EntityFramework;
+using Abp.EntityFrameworkCore.Repositories;
+using Abp.Events.Bus.Entities;
+using Abp.Json;
+using Microsoft.EntityFrameworkCore;
+
+namespace Abp.EntityHistory
+{
+    public class EntitySnapshotManager : IEntitySnapshotManager, ITransientDependency
+    {
+        private readonly IRepository<EntityChange, long> _entityChangeRepository;
+
+        public EntitySnapshotManager(IRepository<EntityChange, long> entityChangeRepository)
+        {
+            _entityChangeRepository = entityChangeRepository;
+        }
+        public async Task<EntityHistorySnapshot> GetSnapshotAsync<TEntity>(int id, DateTime snapshotTime) where TEntity : class, IEntity<int>
+        {
+            return await GetSnapshotAsync<TEntity, int>(id, snapshotTime);
+        }
+        public async Task<EntityHistorySnapshot> GetSnapshotAsync<TEntity, TPrimaryKey>(TPrimaryKey id, DateTime snapshotTime) where TEntity : class, IEntity<TPrimaryKey>
+        {
+            var entity = _entityChangeRepository.GetDbContext()
```

Comment:
can we extract this into 
```csharp
protected virtual async Task<TEntity> GetEntityById<TPrimaryKey>(TPrimaryKey id)
{
    _entityChangeRepository.GetDbContext()
                           .Set<TEntity>()
                           .AsQueryable()
                           .FirstOrDefaultAsync(CreateEqualityExpressionForId(id));
}
```

we use `CreateEqualityExpressionForId` for efcore as well
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/c33224544a18b00512162a489bc55c3ca194431f/src/Abp.EntityFrameworkCore/EntityFrameworkCore/Repositories/EfCoreRepositoryBaseOfTEntityAndTPrimaryKey.cs#L144